WcaOnRails/app/models/team.rb

```diff
@@ -2,4 +2,18 @@ class Team < ActiveRecord::Base
   has_many :team_members, dependent: :destroy
 
   accepts_nested_attributes_for :team_members, reject_if: :all_blank, allow_destroy: true
+
+  validate :membership_periods_cannot_overlap_for_single_user
+  def membership_periods_cannot_overlap_for_single_user
+    team_members.group_by(&:user).each do |user, memberships|
+      memberships.combination(2).to_h.each do |first, second|
```

Comment:
Good catch! I wanted to improve the readability by using block with `first` and `second` arguments instead of an array but you are absolutely right.